### ISSUE TITLE
aria-checked issue for carbon react data table TableSelectAll component when some of the rows are selected.

### ISSUE THREAD

**Issue by sunnyanand17** [IBMa/equal-access#881]
Hi, I am getting below AVT violation for the TableSelectAlll carbon react component , the component in turn calls the InlineCheckbox component . I created the sandbox for this component and see the same issue there as well. https://codesandbox.io/s/elated-joliot-cb4vwr?file=/index.js . The issue is only occurring when not all the rows are selected/unselected but a few are selected. I am not sure how to go about fixing this based on the guidance in the app. The problem is aria-checked is getting set for value mixed and is not happening for no rows selected or all rows selected.

<img width="1680" alt="Screen Shot 2022-05-12 at 1 14 59 PM" src="https://user-images.githubusercontent.com/16520944/168173179-8cd8450f-6dfa-43d4-ae9c-114cfe7e5819.png">

**Comment by shunguoy** [comment]
per [spec](https://w3c.github.io/html-aria/), Authors [MUST NOT use the aria-checked attribute on input type=checkbox elements](https://w3c.github.io/html-aria/#att-checked). Instead, the native 'checked' should be used.

**Comment by shunguoy** [comment]
work as expected, close.  Suggest the user to open an issue with Carbon component to resolve the issue.
